### ISSUE TITLE
Ability to pull service images if newer version available

### ISSUE THREAD

**Issue by dennybaa** [drone/drone#1089]
Hello,

first of all would like to say thank you for such an awesome CI system. I'm going to use it heavily now)
So during my work I face the following issue (as of 0.3.0-alpha0)

Say we have following config:
```yaml
image: quay.io/dennybaa/droneunit-ubuntu
env:
  - BUILDHOST=st2packages-buildenv-deb
script:
  - ./.drone/build.sh
services:
  - quay.io/stackstorm/st2packages-buildenv-deb
```

**Service image update is impossible**

 When I click rebuild or even update the service image this image doesn't get updated during a new triggered build or during manual rebuild. It's looks that current implementation doesn't handle the described situation. 
  
If builds are slow and or if we have parallel builds and service image has been updated in-between runs then we end up referencing the same image (the first one). It's not the best behavior, imo drone should be able to clone service images similar way it does for build images.
However there's an unclear moment if a build was performed with a previous version of service container, and then I decide to rebuild it manually - what version should be used.

Cheers!

**Comment by dennybaa** [comment]
Details: particular tags should be update when service image is pulled,
since doing `docker pull registry_path`  doesn't update the whole family, ie.

`docker pull quay.io/dennybaa/droneunit` is not the same as
`docker pull quay.io/dennybaa/droneunit:trusty-upstart`

**Comment by bradrydzewski** [comment]
marking as started because the 0.4 yaml has a `pull` flag. When `pull: true` drone always downloads the latest image. This works for build images, plugins and service images

**Comment by dennybaa** [comment]
awesome!

**Comment by bradrydzewski** [comment]
marking as closed in 0.4 branch
